modules/dnn/src/layers/padding_layer.cpp

```diff
@@ -58,7 +58,12 @@ class PaddingLayerImpl CV_FINAL : public PaddingLayer
     {
         CV_Assert(inputs.size() == 1);
         const MatShape& inpShape = inputs[0];
+        if (inpShape.empty()){
+            outputs.resize(1, MatShape(1, paddings.size() * 2 + 1));
```

Comment:
Please clarify a bit. If inpShape is empty which means scalar (0D), we should accept only two paddings (before and after). So output shape should be 1D

Reply:
`paddings` is the tuple that contains several paddings from each side. You could pad a scalar twice from both sides. Or three times from left and right.  You get a 1D vector in each case but with a different shape. 1st case [3], second case [5]. Therefore, we need this construct `MatShape(1, paddings.size() * 2 + 1)`

Reply:
Just an example, scalar input (`C`) and paddings has a single pair `(2, 3)`. So output is `[0, 0, C, 0, 0, 0]` and output shape is `[5]`. However code computes output shape as `paddings.size() * 2 + 1 = [3]`.

Reply:
You are right. corrected